projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php

```diff
@@ -362,6 +362,37 @@ function wpcom_global_styles_current_user_can_edit_wp_global_styles( $blog_id =
 	return current_user_can( $wp_global_styles_cpt->cap->publish_posts );
 }
 
+/**
+ * Get the user data from wp global styles.
+ *
+ * We need this wrapper because in rare occasions WP Core triggers a TypeError in this snippet:
+ *
+ * $global_style_query = new WP_Query();
+ * $recent_posts       = $global_style_query->query( $args );
+ * if ( count( $recent_posts ) === 1 ) {
+ * $user_cpt = get_object_vars( $recent_posts[0] );
+ *
+ * In some scenarios, $recent_posts[0] is null and get_object_vars triggers a TypeError.
+ *
+ * @return array
+ */
+function wpcom_get_user_data_from_wp_global_styles() {
+	try {
+		return WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme() );
+	} catch ( TypeError $error ) {
+		wp_clean_theme_json_cache();
+	}
+
+	/**
+	 * Try again after theme cache json is purged. If it's not working, fallback to an empty array.
+	 */
+	try {
+		return WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme() );
+	} catch ( TypeError $error ) {
+		return array();
```

Comment:
Do we need logging? Is it worth it? This happens maybe once a month.

If we add logging, do we need logstash + slack notifications on T-Rex?  Can we configure logstash to ping our team?

Reply:
Yes, via grafana if not in any other way. Logging seems like a good move here so we can understand whether the fix is working.